src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -70,13 +72,40 @@ case object UnknownPeriod extends Period {
 }
 
 case class PeriodSum(periods: Set[Period], modifier: Modifier) extends Period {
-  override def addTo(temporal: Temporal): Temporal = ???
 
-  override def get(unit: TemporalUnit): Long = ???
+  var map = scala.collection.mutable.Map[TemporalUnit,Long]()
 
-  override def subtractFrom(temporal: Temporal): Temporal = ???
+  for ( period <- periods; unit <- period.getUnits.asScala )
+    map.get(unit) match {
+      case Some(value) => map.update( unit, value + period.get(unit))
```

Comment:
You can write this as map(unit) += period.get(unit). http://stackoverflow.com/questions/2662984/what-are-all-the-instances-of-syntactic-sugar-in-scala